src/Package/Impl/Publishing/Handlers/RmdPublishHandler.cs

```diff
@@ -25,12 +30,26 @@ internal sealed class RmdPublishHandler : IMarkdownFlavorPublishHandler {
             return true;
         }
 
-        public string GetCommandLine(string inputFile, string outputFilePath, PublishFormat publishFormat, Encoding encoding) {
-            string format = GetDocumentTypeString(publishFormat);
-            string outputFile = Path.GetFileName(outputFilePath);
-            string outputFolder = Path.GetDirectoryName(outputFilePath).Replace('\\', '/');
-            // Run rmarkdown::render
-            return Invariant($"rmarkdown::render('{inputFile}', output_format='{format}', output_file='{outputFile}',  output_dir='{outputFolder}', encoding='cp{encoding.CodePage}')");
+        public async Task PublishAsync(IRSession session, ICoreShell coreShell,  IFileSystem fs, string inputFilePath, string outputFilePath, PublishFormat publishFormat, Encoding encoding) {
+            try {
+                using(var fts = new FileTransferSession(session, fs)) {
+                    var rmd = await fts.SendFileAsync(inputFilePath);
+                    var result = await RMarkdownRenderAsync(session, rmd, outputFilePath, publishFormat, encoding);
+                    await fts.FetchFileAsync(result, outputFilePath);
```

Comment:
In fact, do we actually derive any benefit from having `RMarkdownRenderAsync` return just the blob ID (with file info), rather than the blob, given that it unconditionally creates a blob anyway? It seems that there's no reason for the caller to call it without calling `FetchFileAsync` later to get the data. And if they forget to call it, they will leave a dangling blob in the host, so potentially it's dangerous, too.

@MikhailArkhipov, what do you think? Is there value in maintaining the split here? Or would it be better to change `RMarkdownRenderAsync` to just take input and output paths (on local FS), and create and use a `FileTransferSession` inside as an implementation detail?